File: liveroom-uikit-core/src/main/java/com/qncube/uikitcore/refresh/CommonEmptyView.java
package com.qncube.uikitcore.refresh;

import android.content.Context;
import android.text.TextUtils;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.annotation.StringRes;

import com.qncube.uikitcore.R;


/**
 * 通用empty 待替换ui设计
 */
public class CommonEmptyView extends FrameLayout implements View.OnClickListener, IEmptyView {

    private boolean clickEnable = true;

    private OnClickListener listener;
    private int mErrorState;
    private String strNoDataContent = "";

    ImageView img;
    TextView emptyText;

    public CommonEmptyView(Context context) {
        super(context);
        init(context);
    }

    public CommonEmptyView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        init(context);
    }

    public CommonEmptyView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context);
    }

    private void init(Context context) {
        LayoutInflater.from(context).inflate(R.layout.kit_view_custom_empty, this);
        setVisibility(GONE);

       // setBackgroundColor(-1);
        setOnClickListener(this);
        img = findViewById(R.id.img);
        emptyText = findViewById(R.id.empty_text);
        img.setOnClickListener(this);
    }

    /**
     * 設置背景
     */
    public void setEmptyIcon(int imgResource) {
        img.setImageResource(imgResource);
    }

    /**
     * 設置内容
     */
    public void setEmptyTips(String noDataContent) {
        strNoDataContent = noDataContent;
        if (emptyText != null) {
            emptyText.setText(strNoDataContent);
        }
    }

    /**
     * 設置内容
     */
    public void setEmptyTips(@StringRes int tipsRes) {
        setEmptyTips(getContext().getString(tipsRes));
    }

    @Override
    public View getContentView() {
        return this;
    }

    /**
     * 根据状态設置当前view
     *
     * @param i
     */
    @Override
    public void setStatus(int i) {

        final boolean disconnected = !NetUtil.isNetworkAvailable(getContext());
        if(refreshingView!=null){
            refreshingView.setVisibility(GONE);
        }
        switch (i) {
            case NETWORK_ERROR:
                setVisibility(View.VISIBLE);
                mErrorState = NETWORK_ERROR;
                emptyText.setText("网络错误");
                img.setImageResource(R.drawable.kit_pic_empty_network);
                img.setVisibility(View.VISIBLE);
                clickEnable = true;
                break;
            case NODATA:
                setVisibility(View.VISIBLE);
                mErrorState = NODATA;
                img.setImageResource(R.drawable.kit_pic_empty);
                img.setVisibility(View.VISIBLE);
                refreshEmptyView();
                clickEnable = true;
                break;
            case HIDE_LAYOUT:
                setVisibility(View.GONE);
                break;
            case START_REFREASH_WHEN_EMPTY:

                if (refreshingView != null) {
                    setVisibility(View.VISIBLE);
                    refreshingView.setVisibility(VISIBLE);
                }
                break;

            default:
                break;
        }
    }

    private View refreshingView = null;

    public void setRefreshingView(View view) {
        refreshingView = view;
        refreshingView.setVisibility(View.GONE);
        addView(refreshingView);

    }

    private void refreshEmptyView() {
        emptyText.setText(TextUtils.isEmpty(strNoDataContent) ?
                "牛儿有泪不轻弹，这就哭一个给你看" : strNoDataContent);

    }

    /**
     * 获取当前错误状态
     *
     * @return
     */
    public int getErrorState() {
        return mErrorState;
    }

    public void setOnLayoutClickListener(OnClickListener listener) {
        this.listener = listener;
    }

    @Override
    public void setVisibility(int visibility) {
        if (visibility == View.GONE) {
            mErrorState = HIDE_LAYOUT;
        }
        super.setVisibility(visibility);
    }

    @Override
    public void onClick(View v) {
        if (clickEnable) {
            if (mErrorState == NETWORK_ERROR) {
             //   Toast.makeText(AppCache.getContext(), "TODO, Network Settings", Toast.LENGTH_LONG);
//                SystemSettingsUtils.launchWirelessSettings(getContext(), true);
                return;
            }
            if (listener != null) {
                listener.onClick(v);
            }
        }
    }
}

File: LibComponent/base_frame/src/main/java/com/hapi/baseframe/smartrecycler/CommonEmptyView.kt
package com.hapi.baseframe.smartrecycler

import android.content.Context
import android.text.TextUtils
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View
import android.widget.FrameLayout
import android.widget.ImageView
import android.widget.TextView
import com.hapi.base_mvvm.R

/**
 * 通用empty 待替换ui设计
 */
class CommonEmptyView : FrameLayout, IEmptyView {
    /**
     * 获取当前错误状态
     *
     * @return
     */
    var errorState = 0
        private set
    private var strNoDataContent = ""
    var img: ImageView? = null
        private set
    var emptyText: TextView? = null
        private set

    constructor(context: Context) : this(context, null)
    constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, 0)
    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    ) {
        init(context)
    }

    private var emptyIcon = R.drawable.kit_pic_empty
    private var emptyNoNetIcon = R.drawable.kit_pic_empty_network
    private fun init(context: Context) {
        LayoutInflater.from(context).inflate(R.layout.view_custom_empty, this)
        visibility = GONE
        // setBackgroundColor(-1);
        img = findViewById(R.id.img)
        emptyText = findViewById(R.id.empty_text)
        setOnClickListener { }
    }

    /**
     * 設置背景
     */
    fun setEmptyIcon(imgResource: Int) {
        emptyIcon = imgResource
        img!!.setImageResource(imgResource)
    }

    fun setEmptyNoNetIcon(imgResource: Int) {
        emptyNoNetIcon = imgResource
    }

    /**
     * 設置内容
     */
    fun setEmptyTips(noDataContent: String) {
        strNoDataContent = noDataContent
        if (emptyText != null) {
            emptyText!!.text = strNoDataContent
        }
    }

    override fun getContentView(): View {
        return this
    }

    /**
     * 根据状态設置当前view
     *
     * @param i
     */
    override fun setStatus(i: Int) {
        if (refreshingView != null) {
            refreshingView!!.visibility = GONE
        }
        when (i) {
            IEmptyView.NETWORK_ERROR -> {
                visibility = VISIBLE
                errorState = IEmptyView.NETWORK_ERROR
                emptyText!!.text = "网络错误"
                img!!.setImageResource(emptyNoNetIcon)
                img!!.visibility = VISIBLE
            }
            IEmptyView.NODATA -> {
                visibility = VISIBLE
                errorState = IEmptyView.NODATA
                img!!.setImageResource(emptyIcon)
                img!!.visibility = VISIBLE
                refreshEmptyView()
            }
            IEmptyView.HIDE_LAYOUT -> visibility = GONE
            IEmptyView.START_REFREASH_WHEN_EMPTY -> if (refreshingView != null) {
                visibility = VISIBLE
                refreshingView!!.visibility = VISIBLE
            }
            else -> {}
        }
    }

    private var refreshingView: View? = null

    fun setRefreshingView(view: View?) {
        refreshingView = view
        refreshingView!!.visibility = GONE
        addView(refreshingView)
    }

    private fun refreshEmptyView() {
        emptyText!!.text = if (TextUtils.isEmpty(strNoDataContent)) "" else strNoDataContent
    }

    override fun setVisibility(visibility: Int) {
        if (visibility == GONE) {
            errorState = IEmptyView.HIDE_LAYOUT
        }
        super.setVisibility(visibility)
    }

}